src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -0,0 +1,212 @@
+        public string Build(SqlUpsertQueryStructure structure)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support updates");
+        }
+
+        /// <summary>
+        /// Add a JSON_QUERY wrapper on the column
+        /// </summary>
+        private string WrapSubqueryColumn(LabelledColumn column, SqlQueryStructure subquery)
+        {
+            string builtColumn = Build(column as Column);
+            if (subquery.IsListQuery)
+            {
+                return $"JSON_QUERY (COALESCE({builtColumn}, '[]'))";
+            }
+
+            return $"JSON_QUERY ({builtColumn})";
+        }
+
+        /// <summary>
+        /// Build columns and wrap columns which represent join subqueries
+        /// </summary>
+        private string WrappedColumns(SqlQueryStructure structure)
+        {
+            return string.Join(", ",
+                structure.Columns.Select(
+                    c => structure.IsSubqueryColumn(c) ?
+                        WrapSubqueryColumn(c, structure.JoinQueries[c.TableAlias!]) + $" AS {QuoteIdentifier(c.Label)}" :
+                        Build(c)
+            ));
+        }
+
+        /// <inheritdoc />
+        public string BuildStoredProcedureResultDetailsQuery(string databaseObjectName)
+        {
+            throw new NotImplementedException("DataWarehouse sql currently does not support stored procedures");
+        }
+
+        /// <summary>
+        /// Builds the query to get all the read-only columns in an DWSql table.
+        /// For DWSql, the columns:
+        /// 1. That have data_type of 'timestamp', or
+        /// 2. are computed based on other columns,
+        /// are considered as read only columns. The query combines both the types of read-only columns and returns the list.
+        /// </summary>
+        /// <param name="schemaOrDatabaseParamName">Param name of the schema/database.</param>
+        /// <param name="tableParamName">Param name of the table.</param>
+        /// <returns>String representing the query needed to get a combined list of read only columns.</returns>
+        public string BuildQueryToGetReadOnlyColumns(string schemaParamName, string tableParamName)
+        {
+            // For 'timestamp' columns sc.is_computed = 0.
+            string query = "SELECT ifsc.column_name from sys.columns as sc INNER JOIN INFORMATION_SCHEMA.COLUMNS as ifsc " +
+                "ON (sc.is_computed = 1 or ifsc.data_type = 'timestamp') " +
+                $"AND sc.object_id = object_id({schemaParamName}+'.'+{tableParamName}) and ifsc.table_name = {tableParamName} " +
+                $"AND ifsc.table_schema = {schemaParamName} and ifsc.column_name = sc.name;";
+
+            return query;
+        }
+
+        /// <inheritdoc/>
+        public string BuildFetchEnabledTriggersQuery()
+        {
+            string query = "SELECT STE.type_desc FROM sys.triggers ST inner join sys.trigger_events STE " +
```

Comment:
Same, is this query to identify triggers are enabled and if yes, does it help us update table with triggers as well - Perhaps, this would come with the support for update..

Reply:
This would be handled in update. same query as mssql with no changes.